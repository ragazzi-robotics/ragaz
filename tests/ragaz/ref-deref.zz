class Bar:
    s: str
    def __init__(self, s: str):
        self.s = s


def main():
    var a = "abc"
    var b = &a
    var c = &b
    var d = *c
    var e = *d
    print(e)

    var a = Bar("def")
    var b = &a
    var c = &b
    print(c.s)

    var a = ["a", "b", "c"]
    var b = &a[1]
    var c = &b
    print(c)

    var a = ("d", "e", "f")
    var b = &a[1]
    var c = &b
    print(c)

    var a1 = 1
    var b1 = &a1
    var c1 = &b1
    var a2 = 2
    var b2 = &a2
    var c2 = &b2
    print(a1 + b1)
