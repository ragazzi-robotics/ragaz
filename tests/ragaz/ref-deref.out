abc
def
b
e
2
